src/core/coap/coap.cpp

```diff
@@ -981,8 +981,13 @@ uint32_t TxParameters::CalculateMaxTransmitWait(void) const
 
 uint32_t TxParameters::CalculateSpan(uint32_t aMaxRetx) const
 {
+#if OPENTHREAD_CONFIG_COAP_API_ENABLE
```

Comment:
This is fine as a quick fix to reduce code size.

Some other suggestion
- Do we really need 64 bit calculation for calculating the coap parameters values? Can we consider using 32 bit in all cases?
- The size increase is coming from configurable coap tx parameters feature. We can cosnider adding a config/build option around it to be able to enable/disable the feature/APIs if not needed (it would reduce code size even more if we really use only constant values and not need to calculate them at run time).

I personally prefer option 1. Thoughts? @bukepo, @jwhui

Reply:
Hi all, I think for normal use cases we could use 32 bit calculations. For very long delays and/or high ReTx count this would result in overflow.

Two comments, besides those by @abtink:

1. We could reverse the order of the division and muliplication, to extend the range a bit.
2. We *should* state clearly this overflow risk in the docs.

Reply:
Yes, this is a quick fix. I  also prefer just using 32bit  for all cases if possible. If that's not possible, I think we need to do all @abtink and @dismirlian suggested.

Reply:
I'm fine with limiting the range if there are no compelling use cases. I agree with @dismirlian that the limitations should be clearly documented.

Reply:
@bukepo, no, it seems fine!

Reply:
If we're worried about OpenThread application code supplying parameters that overflow, one prospect is we change that `#if` to `#if OPENTHREAD_CONFIG_COAP_LONG_TIMEOUT`… which we define in the configuration:

```
/**
 * @def OPENTHREAD_CONFIG_COAP_LONG_TIMEOUT
 *
 * Enable support for long time-out values in CoAP requests.  In ordinary circumstances, the time-out
 * calculations use 32-bit arithmetic which can lead to overflows when computing the total request time
 * span.  Set this to 1 to use 64-bit arithmetic to avoid these overflows.
 */
#ifndef OPENTHREAD_CONFIG_COAP_LONG_TIMEOUT
#define OPENTHREAD_CONFIG_COAP_LONG_TIMEOUT 0
#endif
```